### PR TITLE
gs: add {_strip/unstrip}_protocol

### LINKED ISSUE #38: import-url: importing directory from GCS bucket downloads to parent directory
Originally posted in https://github.com/iterative/dvc/issues/7638

### PR DESCRIPTION
**Pull request by dtrifiro**
fixes resolution of relative paths (https://github.com/iterative/dvc/issues/7638)

fixes #38

### CODE REVIEW

File: src/dvc_objects/fs/implementations/gs.py [hunk 1]
```diff
@@ -23,3 +23,9 @@ def fs(self):
         from gcsfs import GCSFileSystem
 
         return GCSFileSystem(**self.fs_args)
+
+    @classmethod
+    def _strip_protocol(cls, path: str) -> str:
+        from fsspec.utils import infer_storage_options
+
+        return infer_storage_options(path)["path"]
```
Comment (efiop) [inline src/dvc_objects/fs/implementations/gs.py:31]: I'm not sure I follow. In https://github.com/iterative/dvc/pull/7655/files you specifically wrote that we don't need to strip anything and here we are stripping the url down to the path. So what should we do and why?

Comment (dtrifiro) [reply src/dvc_objects/fs/implementations/gs.py:31]: As mentioned in iterative/dvc#7655, only this `_strip_protocol` change is required to fix the import-url issue

Comment (dtrifiro) [reply src/dvc_objects/fs/implementations/gs.py:31]: Got a bit confused by things moving around. Should be fine now, the iterative/dvc#7655 can be closed

**Comment by efiop** [comment]
@dtrifiro Could you please explain the issue and the solution? There is no info on it in the issue, so I'm trying to make sure I understand it correctly

**Review by efiop — APPROVED** [review]
Might want to check other filesystems too. Might have to come back to this later.

**Pull request merged**